client/src/pages/Listing/Listing.tsx

```diff
@@ -105,44 +121,70 @@ const Listing = (): JSX.Element => {
                 type="date"
                 defaultValue="2022-01-01"
                 className={clsx(dateInOff, 'left')}
+                onChange={handleDateInChange}
+                inputRef={dateInRef}
               />
               <TextField
                 id="dateOff"
                 InputProps={{
                   disableUnderline: true,
                   endAdornment: (
                     <InputAdornment position="end">
-                      <IconButton color="primary" aria-label="clear search filter">
-                        <ClearIcon />
-                      </IconButton>
+                      {!filter ? (
+                        <IconButton aria-label="clear/apply search filter" onClick={handleFilterChange}>
+                          <CheckIcon />
+                        </IconButton>
+                      ) : (
+                        <IconButton aria-label="clear/apply search filter" onClick={handleFilterChange}>
+                          <ClearIcon color="primary" />
+                        </IconButton>
+                      )}
                     </InputAdornment>
                   ),
                 }}
                 type="date"
                 defaultValue="2022-01-10"
                 className={clsx(dateInOff, 'right')}
+                onChange={handleDateOffChange}
+                inputRef={dateOffRef}
               />
             </Box>
           </Grid>
         </Grid>
         <Grid container item direction="row" xs={12} md={10}>
           <Grid item container spacing={1}>
-            {sitters.map(({ id, img, fullname, rating, slogan, description, location, price }) => (
-              <ListingItem
-                key={id}
-                image={img}
-                fullName={fullname}
-                rating={rating as rate}
-                slogan={slogan}
-                description={description}
-                location={location}
-                price={price}
-              />
-            ))}
+            {!loading && profiles.length > 0 ? (
+              profiles.map(
+                ({ _id, address, description, firstName, gender, lastName, phoneNumber, rate, availability }) => (
+                  <ListingItem
+                    key={_id}
+                    image={'img'}
+                    fullName={getFullName(firstName, lastName)}
+                    rating={Math.floor(Math.random() * 5) as rate}
+                    slogan={'slogan'}
+                    description={description}
+                    location={address}
+                    price={rate || 0}
```

Comment:
**Suggestion:** maybe consider adding a default value to the 'price' field in the relevant model